cmd/run.go

```diff
@@ -191,6 +191,13 @@ func (r *Runner) Run(ctx context.Context, patterns []string) error {
 		defer server.Stop()
 		r.config.Server = server.FrontendHostPort
 		r.log.Info("Started server", "HostPort", r.config.Server)
+	} else if !strings.Contains(r.config.Server, "tmprl.cloud") {
```

Comment:
This is fickle to rely on string pieces like this. There are other ways to do quick checks on whether namespace is available. I suspect you can change `WaitNamespaceAvailable` to do a never-match list and check for https://pkg.go.dev/go.temporal.io/api/serviceerror#NamespaceNotFound.

What you've found is that we don't have a reasonable server-agnostic way to check if a namespace exists. We should hack one and consider this a pain point for fixing for all users, not just us.

Reply:
Sure, but we've done the other way in tests before and it's got it's own downsides. I agree this should be easier but I don't really think it's in the scope of this PR

Reply:
> I don't really think it's in the scope of this PR

Agree about the general lack of namespace-exists, but this PR adds a brand new `WaitNamespaceAvailable`. Can't we just make that work in all environments? (via hacks or otherwise)

Reply:
I'll just do what we did in TS where you try to describe a nonexistent workflow make sure it says it can't find the workflow rather than the namespace not existing. https://github.com/temporalio/sdk-typescript/blob/e283b893d05ef4777e8ce277413e542ce69ef334/scripts/wait-on-temporal.mjs#L20

Annoying, but, whatever.